x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/ApiKeyBackwardsCompatibilityIT.java

```diff
@@ -333,16 +333,17 @@ private Map<Version, RestClient> getRestClientByVersion() throws IOException {
     }
 
     private static RoleDescriptor randomRoleDescriptor(boolean includeRemoteIndices) {
+        final Set<String> excludedPrivileges = Set.of("cross_cluster_replication", "cross_cluster_replication_internal");
```

Comment:
There is a somewhat esoteric BWC quirk for API keys here:

If we have a mixed cluster with say 8.8 and 8.9 nodes, and we create a (regular) API key that has the `cross_cluster_replication` privilege that hits the 8.9 nodes, things will break for the 8.8 nodes when they try to de-serialize the API key. This is a problem more generally for API keys and new privileges: first discussed [here](https://github.com/elastic/elasticsearch/pull/95512#issuecomment-1526147961). I think it's okay to deal with this separately from this PR though, either in a targeted follow-up or in a more general fix.

Reply:
We could do something about it. But I think it is also OK to leave it as is. I commented [here](https://github.com/elastic/elasticsearch/issues/95771#issuecomment-1533992762)